pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -343,7 +343,9 @@ public static class Broker {
 
     public static final String CONFIG_OF_ENABLE_PARTITION_METADATA_MANAGER =
         "pinot.broker.enable.partition.metadata.manager";
-    public static final boolean DEFAULT_ENABLE_PARTITION_METADATA_MANAGER = false;
+    public static final boolean DEFAULT_ENABLE_PARTITION_METADATA_MANAGER = true;
+    public static final String AUTO_PARTITION_HINT = "pinot.broker.multistage.auto.partition.hint";
```

Comment:
To keep it consistent, name it `CONFIG_OF_AUTO_PARTITION_HINT`.
For other configs, they usually start with a verb, so I personally prefer `autoApplyPartitionHint`. This is optional, but let's keep the default value name and config name consistent

Reply:
I'll change it to `infer.partition.hint`